### ISSUE TITLE
monitor configuration template invalid

### ISSUE THREAD

**Issue by milkotodorov** [naver/ngrinder#195]
In the monitor configuration __agent.conf there is a line:

#monitor.binding_port=hostname_or_ip

which is very confusing as it must be "monitor.binding_ip=...".

**Comment by songeunwoo** [comment]
hi milkotodorov~
Do not be difficult.
When you use this option,
Use when using multiple IP as your target server load balancing.
so never mind.

1. After the move to a folder to install the ngrinder monitor.
2. run_monitor.sh(Linux) or run_monitor.bat &(Windows)
3. check monitor
![2016-08-31 11 27 06](https://cloud.githubusercontent.com/assets/16470009/18113873/efb3cbc6-6f6d-11e6-9e08-08b278fe5097.png)

Pretty simple ^^

Note that the monitor must have specific ports open.
By default 10023.

good luck~

**Comment by milkotodorov** [comment]
hi songeunwoo,

when I start the run_monitor.sh it is binding to the localhost not the IP assigned to the host. As a result the nGrinder controller cannot connect to the monitor. As soon as I use monitor.binding_ip=x.x.x.x it works. However it is initially described as #monitor.binding_port=hostname_or_ip in the __agent.conf

Here is the whole __agent.conf:

```
# If you want to monitor bind to the different local ip not automatically selected ip. Specify below field.
#monitor.binding_port=hostname_or_ip
monitor.binding_port=13243
```
line 2 must be corrected to #monitor.binding_ip=hostname_or_ip isn't it?

Regards,
Milko

**Comment by songeunwoo** [comment]
yes. Public IP right.
i think, please check the ACL on the port 13243.
 ( Controller : Any ==> Monitor : 13243 ) <== When using the monitor port to be opened

to summarize,
1. Sign in to your Controller ngrinder site.
2. Using the Monitor download button-down menu at the top.
3. Unzip the downloaded file monitor.tar
4. run_monitor...

__agent_conf file is nothing to fix.

**Comment by milkotodorov** [comment]
When I try to run the monitor without changing the monitor.binding_ip=x.x.x.x here is how the monitor starts:

```
10:32:28.826 [main] INFO  agent config - NGRINDER_AGENT_HOME : /home/ubuntu/.ngrinder-monitor
10:32:29.015 [main] INFO  org.ngrinder.infra.AgentHome - /home/ubuntu/.ngrinder-monitor/native is created.
10:32:29.029 [main] INFO  starter - ***************************************************
10:32:29.029 [main] INFO  starter - * Start nGrinder Monitor...
10:32:29.029 [main] INFO  starter - ***************************************************
10:32:29.171 [main] INFO  o.n.monitor.agent.RegisterMXBean - Added MXBean:org.ngrinder.monitor:name=System.
10:32:29.172 [main] INFO  o.n.monitor.agent.RegisterMXBean - Default MXBean Initialized.
10:32:29.172 [main] INFO  o.n.monitor.agent.MonitorServer - Service URL:service:jmx:rmi://[::]:13243/jndi/rmi://[::]:13243/jmxrmi is initiated.
10:32:29.227 [main] INFO  o.n.m.collector.DataCollectManager - SystemDataCollector started.
10:32:29.228 [main] INFO  o.n.m.collector.DataCollectManager - Collection interval : 1s).

```

And of course the controller cannot connect to this monitor and they are both running on different hosts.

**Comment by songeunwoo** [comment]
Monitor Controller and the server **must have a port open**.
and so it will not have monitor.binding ip settings.
__agent_conf file is nothing to fix.

**Comment by milkotodorov** [comment]
They do have **all ports** opened.

**Comment by songeunwoo** [comment]
Did you try ping test ?
I will find some more issue.

**Comment by milkotodorov** [comment]
yes - ICMP is also allowed and ping do works between ngrinder controller and the agent and vise versa

**Comment by junoyoon** [comment]
Could you run following command so that we can get to know it's connectable?
```
telnet monitor_ip 13243
```

**Comment by junoyoon** [comment]
And.. did u add the target host(monitoring target) in the test configuration?

**Comment by milkotodorov** [comment]
Hi,

let me describe the scenario a bit more detailed. So nGrinder controller runs on 10.0.86.96 with 1 agent (additionally we have 3 more agents on other 3 VMs).

The monitor is installed on 10.0.65.8. I have ran it without changing anything in its configuration:
![2016-09-05_10-41-55](https://cloud.githubusercontent.com/assets/14890886/18242357/3a3a372a-7356-11e6-94c1-c0aaa9172261.png)

__agent.conf:
![2016-09-05_10-42-16](https://cloud.githubusercontent.com/assets/14890886/18242374/466ebfc0-7356-11e6-8030-a5bc4dfef913.png)

From the nGridner controller VM I can actually connect to the monitor:
![2016-09-05_10-41-16](https://cloud.githubusercontent.com/assets/14890886/18242394/62bde55c-7356-11e6-905b-4a048f156598.png)

However the NGrinder controller seems that it cannot connect to it:
![2016-09-05_10-40-57](https://cloud.githubusercontent.com/assets/14890886/18242407/76850692-7356-11e6-8bd5-336016a4cf8d.png)
![2016-09-05_10-40-46](https://cloud.githubusercontent.com/assets/14890886/18242408/7a8b7f78-7356-11e6-8341-2790b52313e6.png)

The VM with IP 10.0.65.72 has the monitor installed, but the __agent.conf is changed with monitor.binding_ip=10.0.65.72

If I make the same changes on 10.0.65.8 the nGrinder controller connects to this monitor too!
![2016-09-05_10-51-43](https://cloud.githubusercontent.com/assets/14890886/18242513/07159046-7357-11e6-8913-78766ba21199.png)
![2016-09-05_10-53-27](https://cloud.githubusercontent.com/assets/14890886/18242520/0d40a05a-7357-11e6-90b8-18bfb4da6077.png)

I hope this helps to make the picture more clear.

Regards,
Milko

**Comment by junoyoon** [comment]
OK. It's really weird...

We have such a problem before 3.4 was released. 
From 3.4.. the problem was fixed. Could u post the monitor log from the start?

**Comment by milkotodorov** [comment]
Here is the startup log:

09:26:17.325 [main] INFO  agent config - NGRINDER_AGENT_HOME : /home/vcap/.ngrinder_agent
09:26:17.458 [main] INFO  starter - ***************************************************
09:26:17.458 [main] INFO  starter - * Start nGrinder Monitor...
09:26:17.458 [main] INFO  starter - ***************************************************
09:26:17.536 [main] INFO  o.n.monitor.agent.RegisterMXBean - Added MXBean:org.ngrinder.monitor:name=System.
09:26:17.536 [main] INFO  o.n.monitor.agent.RegisterMXBean - Default MXBean Initialized.
09:26:17.536 [main] INFO  o.n.monitor.agent.MonitorServer - Service URL:service:jmx:rmi://[::]:13243/jndi/rmi://[::]:13243/jmxrmi is initiated.
09:26:17.612 [main] INFO  o.n.m.collector.DataCollectManager - SystemDataCollector started.
09:26:17.612 [main] INFO  o.n.m.collector.DataCollectManager - Collection interval : 1s).

Do you need something more?

**Comment by milkotodorov** [comment]
Hello, any update? Just need to replace monitor.binding_port with monitor.binding_ip in the default __agent.conf...

**Comment by wellstyle** [comment]
This part seems to have been confused. 
```
#monitor.binding_port=hostname_or_ip
```

If you set it as follows, it works well.
```
$ vi ~/.ngrinder_agent/agent.conf
monitor.binding_ip=public_ip
$ run_monitor.sh
...
15:26:12.447 [main] INFO  o.n.monitor.agent.MonitorServer - Service URL:service:jmx:rmi://public_ip:13243/jndi/rmi://public_ip:13243/jmxrmi is initiated.
...
```

**Comment by milkotodorov** [comment]
Yes, I figured that out already, but you could fix it so it is no longer confusing for the other people that will eventually face the same issue. 

Again the default configuration for the agent in __agent.conf should *not* 

** monitor.binding_port=hostname_or_ip**

but ** monitor.binding_ip=public_ip ** instead

**Comment by sitmd** [comment]
I have the same issue . When I started my controller at a tomcat , and opened the 8088 port. I runned my ngrinder-monitor at the target server , i got the following log and no target monitor data display in my controller page

13:47:50.282 [main] WARN  agent config - Please run agent with -o option
13:47:50.454 [main] INFO  starter - ***************************************************
13:47:50.454 [main] INFO  starter - * Start nGrinder Monitor... 
13:47:50.454 [main] INFO  starter - ***************************************************
13:47:50.543 [main] INFO  o.n.monitor.agent.RegisterMXBean - Added MXBean:org.ngrinder.monitor:name=System.
13:47:50.544 [main] INFO  o.n.monitor.agent.RegisterMXBean - Default MXBean Initialized.
13:47:50.544 [main] INFO  o.n.monitor.agent.MonitorServer - Service URL:service:jmx:rmi://0.0.0.0:8088/jndi/rmi://0.0.0.0:8088/jmxrmi is initiated.
13:47:50.634 [main] INFO  o.n.m.collector.DataCollectManager - SystemDataCollector started.
13:47:50.635 [main] INFO  o.n.m.collector.DataCollectManager - Collection interval : 1s).
